docs/resources/client/models/modelsystem.md

```diff
@@ -0,0 +1,158 @@
+
+## Common Models and Geometry
+
+The basic model JSON (in `assets/<namespace>/models`) are deserialized into an `UnbakedModel`. The `UnbakedModel` is generally one step short of its baked output, containing some general form of the general properties. The most important thing it contains is the `UnbakedGeometry` via `UnbakedModel#geometry`, which represents the data to become `BakedQuad`s. These quads are inlined into the item and block state model by (eventually) calling `UnbakedGeometry#bake`. This commonly constructs a `QuadCollection`, which contains that list of `BakedQuad`s which can be rendered at anytime or only if a given direction is not culled.
+
+Now, a quad compares to a triangle in a modeling program (and in most other games), however due to Minecraft's general focus on squares, the developers elected to use quads (4 vertices) instead of triangles (3 vertices) for rendering in Minecraft.
+
+:::note
+Models should heavily cache. This is because even though chunks are only rebuilt when a block in them changes, the computations done in this method still need to be as fast as possible and should ideally be cached heavily due to the amount of times this method will be called per chunk section (up to seven times per RenderType used by a given model * amount of RenderTypes used by the respective model * 4096 blocks per chunk section). In addition, [BERs][ber] or [entity renderers][entityrenderer] may actually call this method several times per frame.
+
+This is also true for block state definitions and item models.
+:::
+
+Here are some other methods available within the `UnbakedModel`:
+
+| Signature                          |   Effect                                                                                 |
+|------------------------------------|------------------------------------------------------------------------------------------|
+| `Boolean useAmbientOcclusion()`    | Whether to use [ambient occlusion][ao] or not.                                           |
+| `UnbakedModel.GuiLight guiLight()` | Whether this model renders as 3d (`SIDE`) or flat (`FRONT`) in GUI slots.                |
+| `ItemTransforms transforms()`      | Returns the transforms that can be applied when as an item via its `ItemDisplayContext`. |
+| `TextureSlots.Data textureSlots()` | Returns the texture data that is mapped onto the geometry.                               |
+| `ResourceLocation parent()`        | Returns the parent model that this model will be merged with.                            |
```

Comment:
IIRC this is copied from the model loaders article, and it reads a bit too much as such. The note about caching should go somewhere below, and "Here are some other mods available" is bumpy at best.

Reply:
I've cleaned up the section to explain where the methods are used in the flow of text and moved the cache line to the custom model loaders section.